pkg/storage/s3/s3_test.go

```diff
@@ -0,0 +1,81 @@
+package s3
+
+import (
+	"context"
+	"testing"
+
+	logrTesting "github.com/go-logr/logr/testing"
+
+	velerov1alpha2 "github.com/openshift/managed-velero-operator/pkg/apis/managed/v1alpha2"
+)
+
+func TestSetInstanceBucketName(t *testing.T) {
+	tests := []struct {
+		name            string
+		awsClient       *mockAWSClient
+		bucketName      string
+		matchBucketName bool
+	}{
+		{
+			name:            "set new bucket name in instance status",
+			awsClient:       fakeEmptyClient,
+			bucketName:      "",
+			matchBucketName: false,
+		},
+		{
+			name:            "set recovered bucket name in instance status",
+			awsClient:       fakeClient,
+			bucketName:      "testBucket",
+			matchBucketName: true,
+		},
+		{
+			name:            "don't reclaim inaccessible bucket",
+			awsClient:       fakeInconsistentClient,
+			bucketName:      "inconsistentBucket",
```

Comment:
where is the logic that deals with this differently?

Reply:
The mock AWS client functions only "succeed" for a bucket named `testBucket`. Passing in a bucket named `inconsistentBucket` (or really, anything else), will cause the mock AWS client to return a NotFound awserr.

Reply:
Is this true for your fakeInconsistentClient? You actually have a bucket named "inconsistentBucket". Or is this some magic in AWS' mocking?

Reply:
`fakeInconsistentClient` is a `mockAWSClient` that has been created with an array of (fake) `s3.Bucket`s with a single bucket named "inconsistentBucket". So when `fakeInconsistentClient.ListBuckets()` fires, it returns the list of buckets stored in the client - in this case, one named "inconsistentBucket". But none of the other functions (`GetBucketTagging()`, `GetBucketLocation()`, `HeadBucket()`, etc) return good results for a bucket named anything other than "testBucket".

So when the test has `fakeInconsistentClient` make the call to list its buckets, it returns a list of buckets containing "inconsistentBucket" and then when it tries to list the tags for it or get the its region or try to HEAD the bucket, those functions will return the `NotFound` awserr.

Reply:
Ohhhh okay. I gotcha. Sounds good.